### PR TITLE
Add "no-fips-post" configure option.

### PR DESCRIPTION
**Pull request by slontis**
Using this option disables the OpenSSL FIPS provider self tests.
This is intended for debugging purposes only.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Sticking a breakpoint in an algorithm and then getting confused about what is happening has happened more than once.
It is annoying/time wasting having to step over the self tests.

File: test/recipes/03-test_fipsinstall.t [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env perl
+#usr/bin/env perl
```
Comment (t8m) [inline test/recipes/03-test_fipsinstall.t:1]: This looks wrong

File: INSTALL.md [hunk 2]
```diff
@@ -805,6 +805,13 @@ Build (and install) the FIPS provider
 Don't perform FIPS module run-time checks related to enforcement of security
 parameters such as minimum security strength of keys.
 
+### no-fips-post
+
+Dont perform FIPS module Power On Self Tests.
+
+This option should only be used for debugging only. It is useful when trying to
```
Comment (t8m) [inline INSTALL.md:812]: I would use stronger words: `This option must be used for debugging only as it makes the fips provider non-compliant. It is useful when trying to add breakpoints when testing FIPS algorithms.`

**Comment by paulidale** [comment]
Would changing the FIPS provider's name to something including _non-compliant_ be prudent too?

**Comment by t8m** [comment]
> Would changing the FIPS provider's name to something including _non-compliant_ be prudent too?

Not name, that would extremely complicate things, but description could be changed.

**Comment by slontis** [comment]
Changed the provider description for this option

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**